src/assets/js/ledyer-payments.js

```diff
@@ -0,0 +1,274 @@
+
+                // redirect the user to a success page
+            }
+        } catch ( error ) {
+            unblockUI()
+        }
+    }
+
+    const printNotice = ( message ) => {
+        const elementId = `${ gatewayId }-error-notice`
+
+        // Remove any existing notice that we have created. This won't remove the default WooCommerce notices.
+        $( `#${ elementId }` ).remove()
+
+        const html = `<div id='${ elementId }' class='woocommerce-NoticeGroup'><ul class='woocommerce-error' role='alert'><li>${ message }</li></ul></div>`
+        $( "form.checkout" ).prepend( html )
+
+        document.getElementById( elementId ).scrollIntoView( { behavior: "smooth" } )
+    }
+
+    const logToFile = ( message, level = "notice" ) => {
+        const { logToFileUrl, logToFileNonce, reference } = gatewayParams
+        console.debug( message )
+
+        $.ajax( {
+            url: logToFileUrl,
+            type: "POST",
+            dataType: "json",
+            data: {
+                level,
+                reference,
+                message: message,
+                nonce: logToFileNonce,
+            },
+        } )
+    }
+
+    const unblockUI = () => {
+        $( ".woocommerce-checkout-review-order-table" ).unblock()
+        $( "form.checkout" ).removeClass( "processing" ).unblock()
+    }
+
+    const blockUI = () => {
+        /* Order review. */
+        $( ".woocommerce-checkout-review-order-table" ).block( {
+            message: null,
+            overlayCSS: {
+                background: "#fff",
+                opacity: 0.6,
+            },
+        } )
+
+        $( "form.checkout" ).addClass( "processing" )
+        $( "form.checkout" ).block( {
+            message: null,
+            overlayCSS: {
+                background: "#fff",
+                opacity: 0.6,
+            },
+        } )
+    }
+
+    const isActiveGateway = () => {
+        if ( $( 'input[name="payment_method"]:checked' ).length ) {
+            const currentGateway = $( 'input[name="payment_method"]:checked' ).val()
+            return currentGateway.indexOf( gatewayId ) >= 0
+        }
+
+        return false
+    }
+
+    /**
+     * Update the nonce values.
+     *
+     * This is required when a guest user is logged in and the nonce values are updated since the nonce is associated with the user ID (0 for guests).
+     *
+     * @param {object} nonce An object containing the new nonce values.
+     */
+    const updateNonce = ( nonce ) => {
+        for ( const key in nonce ) {
+            if ( key in gatewayParams ) {
+                gatewayParams[ key ] = nonce[ key ]
+            }
+        }
+    }
+
+    const submitOrderFail = ( error, message ) => {
+        console.debug( "[%s] Woo failed to create the order. Reason: %s", error, message )
+
+        printNotice( message )
+        unblockUI()
+        $( document.body ).trigger( "checkout_error" )
+        $( document.body ).trigger( "update_checkout" )
```

Comment:
printNotice followed by update_checkout will clear the notice you printed on updated_checkout.